types/types.go

```diff
@@ -1008,10 +1008,20 @@ func ValidateCPUReservationValues(instanceManagerCPUStr string) error {
 	if err != nil {
 		return errors.Wrapf(err, "invalid guaranteed/requested instance manager CPU value (%v)", instanceManagerCPUStr)
 	}
-	isUnderLimit := instanceManagerCPU < 0
-	isOverLimit := instanceManagerCPU > 40
-	if isUnderLimit || isOverLimit {
-		return fmt.Errorf("invalid requested instance manager CPUs. Valid instance manager CPU range between 0%% - 40%%")
+
+	switch instanceManagerCPUStr {
+	case SettingDefinitionGuaranteedInstanceManagerCPU.Default:
+		isUnderLimit := instanceManagerCPU < 0
+		isOverLimit := instanceManagerCPU > 40
+		if isUnderLimit || isOverLimit {
+			return fmt.Errorf("invalid requested v1 data engine instance manager CPUs. Valid instance manager CPU range between 0%% - 40%%")
+		}
+	case SettingDefinitionV2DataEngineGuaranteedInstanceManagerCPU.Default:
+		isUnderLimit := instanceManagerCPU < 0
+		isOverLimit := instanceManagerCPU > 2000
+		if isUnderLimit || isOverLimit {
+			return fmt.Errorf("invalid requested v2 data engine instance manager CPUs. Valid instance manager CPU range between 0 - 2000 millicpu")
+		}
```

Comment:
Why passing `instanceManagerCPUStr`? Don't we need to pass the setting `instanceManagerCpuSetting` instead? Then, we are able to do a different range check for it.

In the current code, if `instanceManagerCPUStr` is `20`, there is no `case` able to handle it?

Reply:
> Why passing instanceManagerCPUStr? Don't we need to pass the setting instanceManagerCpuSetting instead?

Yes, passing `instanceManagerCpuSetting` is also feasible.  I reuse the existing function here, but we can improve it in another ticket later.

> In the current code, if instanceManagerCPUStr is 20, there is no case able to handle it?

For an acceptable value, the function returns nil error and the value will be updated in https://github.com/longhorn/longhorn-manager/blob/master/datastore/longhorn.go#L112